pkg/controller/daemon/daemoncontroller.go

```diff
@@ -1144,20 +1144,13 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 		case *predicates.PredicateFailureError:
 			var emitEvent bool
 			switch reason {
-			// unintentional predicates reasons need to be fired out to event.
+			// unexpected
 			case
 				predicates.ErrDiskConflict,
 				predicates.ErrVolumeZoneConflict,
 				predicates.ErrMaxVolumeCountExceeded,
 				predicates.ErrNodeUnderMemoryPressure,
-				predicates.ErrNodeUnderDiskPressure:
-				// wantToRun and shouldContinueRunning are likely true here. They are
-				// absolutely true at the time of writing the comment. See first comment
-				// of this method.
-				shouldSchedule = false
-				emitEvent = true
```

Comment:
Just curious why want this? Seems this predicate error is unexpected and the code should never go here? ;-)